pkg/harvester/components/SettingList.vue

```diff
@@ -0,0 +1,208 @@
+<script>
+import { mapGetters } from 'vuex';
+import { Banner } from '@components/Banner';
+import AsyncButton from '@shell/components/AsyncButton';
+import { HCI_ALLOWED_SETTINGS, HCI_SETTING } from '../config/settings';
+
+const CATEGORY = {
+  ui: [
+    'branding',
+    'ui-source',
+    'ui-plugin-index',
+    'ui-index',
+  ]
+};
+
+export default {
+  name: 'SettingLists',
+
+  components: {
+    AsyncButton,
+    Banner,
+  },
+
+  props: {
+    settings: {
+      type:     Array,
+      required: true,
+    },
+
+    initSettings: {
+      type:     Array,
+      required: true,
+    },
+
+    category: {
+      type:     String,
+      required: true,
+    }
+  },
+
+  data() {
+    const categorySettings = this.settings.filter((s) => {
+      if (this.category !== 'advanced') {
+        return (CATEGORY[this.category] || []).find(item => item === s.id);
+      } else if (this.category === 'advanced') {
+        const allCategory = Object.keys(CATEGORY);
+
+        return !allCategory.some(category => (CATEGORY[category] || []).find(item => item === s.id));
+      }
+    }) || [];
+
+    return {
+      HCI_SETTING,
+      categorySettings,
+    };
+  },
+
+  computed: { ...mapGetters({ t: 'i18n/t' }) },
+
+  methods: {
+    showActionMenu(e, setting) {
+      const actionElement = e.srcElement;
+
+      this.$store.commit(`action-menu/show`, {
+        resources: setting.data,
+        elem:      actionElement
+      });
+    },
+
+    getSettingOption(id) {
+      return HCI_ALLOWED_SETTINGS.find(setting => setting.id === id);
+    },
+
+    toggleHide(s) {
+      this.initSettings.find((setting) => {
```

Comment:
It should be changed to `this.settings.find`.   Otherwise, the `hide` and `show` operations do not take effect